src/datepicker/datepicker.js

```diff
@@ -629,7 +629,17 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
         scope.$parent.$watch(getAttribute, function(value) {
           scope.watchData[key] = value;
           if (key === 'minDate' || key === 'maxDate') {
-            cache[key] = new Date(value);
+            var date = parseDate(value);
+            if (!date) {
+              date = new Date(value);
+              if (angular.isDate(date) && !isNaN(date)) {
+                var dateArray = date.toJSON().split('T')[0].split('-');
```

Comment:
As stated prior, one can get the information from the date via this mechanism:

```js
var year = date.getFullYear(),
  month = date.getMonth(),
  day = date.getDate();
```
This is a better way of getting the date information